homeassistant/helpers/storage.py

```diff
@@ -286,20 +288,36 @@ def async_delay_save(
             "data_func": data_func,
         }
 
+        next_when = self.hass.loop.time() + delay
+        if self._delay_handle and self._delay_handle.when() < next_when:
```

Comment:
Wouldn't `self._delay_handle.when() < next_when` always be `True` because we just calculated it based on latest time?

Reply:
Its for if they call `async_save_delay` again with a shorter delay than the first time